### PR TITLE
Bump to Linux 6.2, gcc12

### PR DESCRIPTION
**Pull request by jerryz123**
- Bump to linux 6.2 with SBI 1.2, busybox 1.36, gcc12.2
- Stop supporting BBL
- Use UART for earlycon
- Added new directory layer to differentiate between builds for different board targets.
- Fixes bug with correct busybox not being used for initramfs targets.

### CODE REVIEW

**Comment by jerryz123** [comment]
Everything should work sans fedora (which was already broken) and some of the marshal test features

File: .github/workflows/run-tests.yml [hunk 1]
```diff
@@ -63,7 +71,7 @@ jobs:
           cd ${{ env.REMOTE_WORK_DIR }}
           conda env create -f ./conda-reqs.yaml -p ./.conda-env
           eval "$(conda shell.bash hook)"
-          conda install -y -p $PWD/.conda-env -c ucb-bar riscv-tools
+          conda install -y -p $PWD/.conda-env -c ucb-bar riscv-tools=1.0.4
```
Comment (abejgonzalez) [inline .github/workflows/run-tests.yml:74]: Does this need to be updated in the CY conda env?

File: boards/firechip/base-workloads/br-base/overlay/etc/init.d/rcS [hunk 2]
```diff
@@ -0,0 +1,27 @@
+#!/bin/sh
+
+
+# Start all init scripts in /etc/init.d
```
Comment (abejgonzalez) [inline boards/firechip/base-workloads/br-base/overlay/etc/init.d/rcS:4]: What is this needed for?

File: docs/source/internal/Build.rst [hunk 3]
```diff
@@ -92,8 +92,8 @@ Linux Kernel Generation and Linking
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 With all of the dependencies finished, we can finally compile the Linux kernel
 and link it with the bootloader. While each workload can use a custom kernel
-source, all workloads use the same bootloader (for now), located at
-``riscv-pk/``. The final linked bbl+linux+initramfs is coppied into
+source, all workloads use the same bootloader.
+The final linked sbi+linux+initramfs is coppied into
```
Comment (abejgonzalez) [inline docs/source/internal/Build.rst:96]:
```suggestion
The final linked sbi+linux+initramfs is copied into
```

File: images/.gitignore [hunk 4]
```diff
@@ -1,3 +0,0 @@
-# This is just a working directory meant to hold images and binaries that shouldn't be tracked by git
-*
-!.gitignore
```
Comment (abejgonzalez) [inline images/.gitignore:3]: Keep this no?

File: wlutil/config.py [hunk 5]
```diff
@@ -488,9 +472,9 @@ def __init__(self, cfgFile=None, cfgDict=None):
                 self.cfg['firesim']['simulation_inputs'] = [pathlib.Path(f) for f in self.cfg['firesim']['simulation_inputs']]
 
         if 'out-dir' in self.cfg:
-            self.cfg['out-dir'] = wlutil.getOpt('image-dir') / self.cfg['out-dir']
+            self.cfg['out-dir'] = wlutil.getOpt('image-dir') / wlutil.getOpt('board-dir').name / self.cfg['out-dir']
```
Comment (abejgonzalez) [inline wlutil/config.py:475]: Did you add an extra folder layer to the images? This `board-dir` to further distinguish images.

File: wlutil/wlutil.py [hunk 6]
```diff
@@ -426,6 +420,8 @@ def initialize():
     for d in initramfs_disk_dirs:
         if not (ctx['initramfs-dir'] / 'disk' / d).exists():
             (ctx['initramfs-dir'] / 'disk' / d).mkdir(parents=True)
+        if not (ctx['initramfs-dir'] / 'nodisk' / d).exists():
```
Comment (abejgonzalez) [inline wlutil/wlutil.py:423]: What is this `nodisk` that I see now?

**Review by abejgonzalez — CHANGES_REQUESTED** [review]
I see that fedora running is commented out. Is that intentional? Also does this solve the network booting issue in Spike?

File: .github/workflows/run-tests.yml (hunk 1 above)
Comment (jerryz123) [reply .github/workflows/run-tests.yml:74]: 1.0.3 and 1.0.4 should be near equivalent.

File: boards/firechip/base-workloads/br-base/overlay/etc/init.d/rcS (hunk 2 above)
Comment (jerryz123) [reply boards/firechip/base-workloads/br-base/overlay/etc/init.d/rcS:4]: This fixes the hang after "Starting Network" for some reason. Its the same as the default script, but with a print statement added.

Its not really a fix, but I have noticed this making that issue go away with this change,

File: images/.gitignore (hunk 4 above)
Comment (jerryz123) [reply images/.gitignore:3]: OOps

File: wlutil/config.py (hunk 5 above)
Comment (jerryz123) [reply wlutil/config.py:475]: Yes I'll add a note on this to the PR

File: wlutil/wlutil.py (hunk 6 above)
Comment (jerryz123) [reply wlutil/wlutil.py:423]: Marshal creates separate `disk` and `nodisk` sysroots for disk/initramfs builds. Previously, the requested busybox was only being copied into the `disk` side, which meant that initramfs builds did not use the correct busybox.

This fixes that.

**Comment by jerryz123** [comment]
> I see that fedora running is commented out. Is that intentional? Also does this solve the network booting issue in Spike?

Yes, the fedora build time is 1.5 hours in qemu for some reason... no idea why, some service takes ages to bring up.

The extra print statements in the init script seem to prevent the issue from manifesting in spike... not sure about RTL sims.

**Comment by abejgonzalez** [comment]
> > I see that fedora running is commented out. Is that intentional? Also does this solve the network booting issue in Spike?
> 
> Yes, the fedora build time is 1.5 hours in qemu for some reason... no idea why, some service takes ages to bring up.
> 
> The extra print statements in the init script seem to prevent the issue from manifesting in spike... not sure about RTL sims.

To be clear, building Fedora is fine... but running it takes too long in QEMU? So basically `guest-init` in Fedora is effectively broken?

**Review by abejgonzalez — DISMISSED** [review]
LGTM pending the CI fixes

**Review by abejgonzalez — APPROVED** [review]
LGTM. We will add Ubuntu as an image later (aka Fedora taking a long time is fine for now)

**Pull request merged**